extensions/couchbase/processors/PutCouchbaseKey.h

```diff
@@ -0,0 +1,161 @@
+#include "core/ProcessSession.h"
+#include "utils/Enum.h"
+#include "core/logging/LoggerConfiguration.h"
+#include "CouchbaseClusterService.h"
+#include "couchbase/persist_to.hxx"
+#include "couchbase/replicate_to.hxx"
+
+namespace magic_enum::customize {
+
+template <>
+constexpr customize_t enum_name<::couchbase::persist_to>(::couchbase::persist_to value) noexcept {
+  switch (value) {
+    case ::couchbase::persist_to::none:
+      return "NONE";
+    case ::couchbase::persist_to::active:
+      return "ACTIVE";
+    case ::couchbase::persist_to::one:
+      return "ONE";
+    case ::couchbase::persist_to::two:
+      return "TWO";
+    case ::couchbase::persist_to::three:
+      return "THREE";
+    case ::couchbase::persist_to::four:
+      return "FOUR";
+  }
+  return invalid_tag;
+}
+
+template <>
+constexpr customize_t enum_name<::couchbase::replicate_to>(::couchbase::replicate_to value) noexcept {
+  switch (value) {
+    case ::couchbase::replicate_to::none:
+      return "NONE";
+    case ::couchbase::replicate_to::one:
+      return "ONE";
+    case ::couchbase::replicate_to::two:
+      return "TWO";
+    case ::couchbase::replicate_to::three:
+      return "THREE";
+  }
+  return invalid_tag;
+}
+}  // namespace magic_enum::customize
+
+namespace org::apache::nifi::minifi::couchbase::processors {
+
+class PutCouchbaseKey final : public core::AbstractProcessor<PutCouchbaseKey> {
+ public:
+  using core::AbstractProcessor<PutCouchbaseKey>::AbstractProcessor;
+
+  EXTENSIONAPI static constexpr const char* Description = "Put a document to Couchbase Server via Key/Value access.";
+
+  EXTENSIONAPI static constexpr auto CouchbaseClusterControllerService = core::PropertyDefinitionBuilder<>::createProperty("Couchbase Cluster Controller Service")
+      .withDescription("A Couchbase Cluster Controller Service which manages connections to a Couchbase cluster.")
+      .withAllowedTypes<controllers::CouchbaseClusterService>()
+      .isRequired(true)
+      .build();
+  EXTENSIONAPI static constexpr auto BucketName = core::PropertyDefinitionBuilder<>::createProperty("Bucket Name")
+      .withDescription("The name of bucket to access.")
+      .withDefaultValue("default")
+      .isRequired(true)
+      .supportsExpressionLanguage(true)
+      .build();
+  EXTENSIONAPI static constexpr auto ScopeName = core::PropertyDefinitionBuilder<>::createProperty("Scope Name")
+      .withDescription("Scope to use inside the bucket. If not specified, the _default scope is used.")
+      .supportsExpressionLanguage(true)
+      .build();
+  EXTENSIONAPI static constexpr auto CollectionName = core::PropertyDefinitionBuilder<>::createProperty("Collection Name")
+      .withDescription("Collection to use inside the bucket scope. If not specified, the _default collection is used.")
+      .supportsExpressionLanguage(true)
+      .build();
+  EXTENSIONAPI static constexpr auto DocumentType = core::PropertyDefinitionBuilder<3>::createProperty("Document Type")
```

Comment:
I think we should use `enum_count` in cases like this, mainly for readability
```suggestion
  EXTENSIONAPI static constexpr auto DocumentType = core::PropertyDefinitionBuilder<magic_enum::enum_count<CouchbaseValueType>()>::createProperty("Document Type")
```

Reply:
Updated in https://github.com/apache/nifi-minifi-cpp/pull/1875/commits/e99afe0d8127c14de5379fbd0f5ee8da7f2ace9c